midend/parserUnroll.h

```diff
@@ -150,16 +189,94 @@ class ParserRewriter : public PassManager {
 // The following are applied to the entire program
 
 class RewriteAllParsers : public Transform {
-    ReferenceMap* refMap;
-    TypeMap*      typeMap;
-    bool          unroll;
+    ReferenceMap*           refMap;
+    TypeMap*                typeMap;
+    bool                    unroll;
+    const IR::ParserState*  outOfBoundsState;
+    const IR::Type_Declaration* stdMetadataType;
+    const IR::Type_Declaration* error;
+
  public:
     RewriteAllParsers(ReferenceMap* refMap, TypeMap* typeMap, bool unroll) :
-            refMap(refMap), typeMap(typeMap), unroll(unroll)
+            refMap(refMap), typeMap(typeMap), unroll(unroll), outOfBoundsState(nullptr),
+            stdMetadataType(nullptr), error(nullptr)
     { CHECK_NULL(refMap); CHECK_NULL(typeMap); setName("RewriteAllParsers"); }
+
+    // generate OutOfBound state
+    const IR::Node* preorder(IR::P4Program *p4Program) override {
+        // getting error variable
+        const auto errorEnum = p4Program->getDeclsByName("error");
+        BUG_CHECK(errorEnum, "Can't find error variable");
+        const auto errorVector = errorEnum->toVector();
+        BUG_CHECK(errorVector->size(), "Can't find error variable %1%", errorVector);
+        const auto error0 = errorVector->at(0);
+        BUG_CHECK(error0->is<IR::Type_Declaration>(), "Can't translate error to expression");
+        error = error0->to<IR::Type_Declaration>();
+
+        const auto stdMetadataTypeEnum = p4Program->getDeclsByName("standard_metadata_t");
+        BUG_CHECK(stdMetadataTypeEnum, "Can't find standard_metadata_t type");
+        const auto stdMetadataTypeVector = stdMetadataTypeEnum->toVector();
+        BUG_CHECK(stdMetadataTypeVector->size(), "Can't find standard_metadata_t type");
+        BUG_CHECK(stdMetadataTypeVector->at(0)->is<IR::Type_Declaration>(),
+                  "Can't translate standard_metadata_t type");
+        stdMetadataType = stdMetadataTypeVector->at(0)->to<IR::Type_Declaration>();
+
+        return p4Program;
+    }
+
+    // start generation of a code
     const IR::Node* postorder(IR::P4Parser* parser) override {
+        BUG_CHECK(stdMetadataType, "The standard_metadata_t type was not found");
+
+        // getting name of standard_metadata_t parameter
```

Comment:
A parser does not need to have a standard_metadata_t parameter.
This is a feature of the architecture.

Reply:
You cannot assign to the error field directly.
What you should do instead is call the function `verify(false, error.OutOfBounds)`